ingress/controllers/nginx/controller.go

```diff
@@ -217,6 +257,75 @@ func (lbc *loadBalancerController) getUDPConfigMap(ns, name string) (*api.Config
 	return lbc.client.ConfigMaps(ns).Get(name)
 }
 
+func (lbc *loadBalancerController) updateEpNamedPorts(key string) {
+	if !lbc.controllersInSync() {
+		lbc.svcEpQueue.requeue(key, fmt.Errorf("deferring sync till endpoints controller has synced"))
+		return
+	}
+
+	svcObj, svcExists, err := lbc.svcLister.GetByKey(key)
+	if err != nil {
+		glog.Warningf("error getting service %v: %v", key, err)
+		return
+	}
+
+	if !svcExists {
+		glog.Warningf("service %v not found", key)
+		return
+	}
+
+	svc := svcObj.(*api.Service)
+	if svc.Spec.Selector == nil {
+		return
+	}
+
+	pods, err := lbc.client.Pods(svc.Namespace).List(api.ListOptions{
```

Comment:
It's still a list of all pods between etcd and the apiserver. Also you're only decreasing the chances of oom, because whose to say you're not running a single svc with 30 pods per node on a 1000 node cluster. I'd vote for:
1. List in a very rare case, if you can detect exactly when it's required
2. If you're listing on "every X update/add/delete", then use a watch

If you want to put that in a todo it's fine. Please move the listing out into a funtion called ListAllPods so we can easily swap it out when we refactor.